### PR TITLE
[8] Proof of Concept Form

### PR DESCRIPTION
**Pull request by julia-rom**
# Description

This is meant to represent the Site Details and Location Form (first tab/section). It integrates MUI,  React Hook Form, and Yup validation.

Fixes # [(8)](https://github.com/globalmangrovewatch/gmw-users/issues/8)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Instructions

1. Npm i
2. Npm start
3. Test out filling out form
4. should see validation messaging for required sections

# How Has This Been Tested?

Fill out each section of the form, miss required section, and try toggling project start/end date

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by julia-rom** [comment]
so I've been stuck on the same issue for a while and hoping I can get a second pair of eyes on this! 

the project end date section should be hidden when 'Does the project have an end date?' has a value of 'false'.  This works in the initial load (it is hidden), and the default value is false. When you switch to true, it appears, however, if you try to switch back to false, it remains in the UI. 

I am using the react hook form `watch ` function and logging the output which confirms the value is switching from true to false. I followed [this example.](https://react-hook-form.com/api/useform/watch)

I also found[ this code sandbox](https://codesandbox.io/s/react-hook-form-watch-v7-qbxd7?file=/src/index.js:1567-1582) which demonstrates a similar example.

on line 39, I have `const watchHasProjectEndDate = watch('hasProjectEndDate', false)` and a conditional JSX section here:


Screen Shot 2022-04-27 at 2 45 56 PM

I am stumped why the section remains, even when the 'Does the project have an end date?' section is set to false.

@JBurkinshaw - maybe you can take a look with a fresh set of eyes!

**Comment by julia-rom** [comment]
stuck on a new issue - having trouble getting the controller wrap for autocomplete to work properly. Seeing this error when I click on a country now : 
Screen Shot 2022-04-28 at 3 26 08 PM

the code below... it fails with the controller, but works on it's own (without being registered through the controller) in the commented out code. 

Screen Shot 2022-04-28 at 3 27 37 PM

been trying things out for an hour. Going to take a look at the api issue for some context change. If you get some time, maybe you can take a look @JBurkinshaw ?

**Comment by julia-rom** [comment]
I set this to ready for review since I believe it meets the requirements of the current task. It integrates Yup, React hook form, and MUI for all the questions of the first section excluding the map. 

It does *not* include an API call, which will be added in the next section. It would be good to test the validation here for each part. The submit button console logs all the form data.

File: mrtt-ui/package.json [hunk 1]
```diff
@@ -3,20 +3,28 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@emotion/react": "^11.9.0",
+    "@hookform/resolvers": "^2.8.8",
     "@mui/material": "^5.6.3",
     "@mui/styled-engine-sc": "^5.6.1",
+    "@mui/x-date-pickers": "^5.0.0-alpha.2",
```
Comment (JBurkinshaw) [inline mrtt-ui/package.json:10]: Is the alpha release the preferred option here? Ideally we should be sticking to stable releases.

File: mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
+// import { useEffect } from 'react'
```
Comment (JBurkinshaw) [inline mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js:1]: Can you remove this if it's not being used please?

File: mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  TextField,
+  Checkbox,
+  FormLabel,
+  InputLabel,
+  RadioGroup,
+  Radio,
+  FormControlLabel,
+  Button,
+  Stack
+} from '@mui/material'
```
Comment (JBurkinshaw) [inline mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js:17]: This is a personal preference nit-pick but I like to keep imports alphabetical

File: mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    }),
+    countries: Yup.array()
+      .of(
+        Yup.object().shape({
+          name: Yup.string(),
+          code: Yup.string()
+        })
+      )
+      .min(1)
+      .typeError('Select at least one item')
```
Comment (JBurkinshaw) [inline mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js:42]: Can you be specific with what needs to be selected?
```suggestion
      .typeError('Select at least one country')
```

File: mrtt-ui/package.json [hunk 5]
```diff
@@ -3,20 +3,28 @@
...
     "@types/node": "^16.11.27",
     "@types/react": "^18.0.5",
     "@types/react-dom": "^18.0.0",
+    "date-fns": "^2.28.0",
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
+    "react-hook-form": "^7.30.0",
+    "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
+    "sass": "^1.51.0",
```
Comment (JBurkinshaw) [inline mrtt-ui/package.json:24]: We should avoid sass if we're using styled-components

File: mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+  const options = ['Restoration/Rehabilitation', 'Afforestation', 'Protection', 'Other']
+
+  return (
+    <div className={styles.projectDetailsForm}>
+      <h1>Project Details Form</h1>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        {/* Project Title */}
+        <div className={styles.formGroup}>
+          <InputLabel sx={{ color: 'black' }}>1.1 Project title</InputLabel>
```
Comment (JBurkinshaw) [inline mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js:65]: Could this styling that can happen at a global level with a [custom theme](https://mui.com/material-ui/customization/theming/) file? (associated issue: #21)

File: mrtt-ui/src/views/Home.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+// import '../styles/Sample.css'
```
Comment (JBurkinshaw) [inline mrtt-ui/src/views/Home.js:2]: Please delete

**Review by JBurkinshaw — CHANGES_REQUESTED** [review]
Nice work. Looks good for the most part.

With regard to styling, can we avoid using sass to keep things simpler?
- A theme can be set for MUI as described in #21.
- styled-components can be used for more granular control of specific components/elements: https://mui.com/system/styled/
- A top-level CSS file can be used for layout and anything else we need if for.

File: mrtt-ui/package.json (hunk 1 above)
Comment (julia-rom) [reply mrtt-ui/package.json:10]: Yep I can get rid of sass and set a theme for MUI. With regards to date-pickers release, this is the latest ver
Screen Shot 2022-05-02 at 9 06 55 AM
sion. The last three versions all have alpha in them... first one has 177k + downloads, maybe a safer bet?

File: mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js (hunk 3 above)
Comment (julia-rom) [reply mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js:17]: sure, will alphabetize

File: mrtt-ui/package.json (hunk 5 above)
Comment (julia-rom) [reply mrtt-ui/package.json:24]: will remove!

File: mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js (hunk 6 above)
Comment (julia-rom) [reply mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js:65]: yep, I added a comment to issue #21 - should include removing sass + adding custom theme file so we can separate those tasks with this PR if that works!

File: mrtt-ui/package.json (hunk 1 above)
Comment (JBurkinshaw) [reply mrtt-ui/package.json:10]: Sound like its the [direction they're taking](https://www.npmjs.com/package/@mui/x-date-pickers) with MUI. Let's just stick to the latest with the aim of switching to a stable release when possible.

**Review by JBurkinshaw — APPROVED** [review]

**Pull request merged**